packages/devtools_app/lib/src/split.dart

```diff
@@ -102,13 +108,42 @@ class _SplitState extends State<Split> {
     return LayoutBuilder(builder: _buildLayout);
   }
 
+  @override
+  void dispose() {
+    preferences.splitFractions.removeListener(onChangeSplitFractions);
+    super.dispose();
+  }
+
+  void onChangeSplitFractions() {
+    setState(() {
```

Comment:
add a helper to perform all the json decoding and updating fractions.  That logic is currently duplicated and we probably want to use a slightly different decoding scheme + check to make sure the decoded fractions aren't identical to the existing fractions. That would avoid possible infinite loops repeatedly decoding and encoding and performing extra rebuilds.